### DISCUSSION TITLE
WCSS 5.x & Component Updates

### DISCUSSION THREAD

**Discussion by tfales** [AlaskaAirlines/WC-Generator#519] [Announcements]
Hi folks! I'm working through a dependency audit on RAIN UI. We have a good handful of auro-related updates, but I think I'm blocked from updating any of these. If I'm understanding this correctly, I can't update WCSS to 5.x because we're utilizing some components that don't have compatible versions yet? And without ability to update WCSS to 5.x, I think that prevents me from updating a bunch of other, compatible components. 

Here's a list of our outdated packages:
![image](https://github.com/AlaskaAirlines/WC-Generator/assets/86446469/7dd77a4d-bfd9-47dd-97b7-af027903597b)

When I try to run `npm i @aurodesignsystem/webcorestylesheets@latest` I get the following error:
![image](https://github.com/AlaskaAirlines/WC-Generator/assets/86446469/71f66d2f-43c2-40b9-8de9-3d78f72aa10c)

I've tried a handful of install command variations, running all packages individually and trying to group the installs together. Any thoughts on how I can go about getting some/all of these packages updated? 

@bdahl1 for reference.

**Comment by blackfalcon** [comment]
Hi @tfales 

I don't think that you are blocked, but I can see where you are confused. The WCSS 5.x update was primarily around updating the tokens reference from `--auro-` to `--ds-` https://github.com/AlaskaAirlines/WebCoreStyleSheets/compare/v4.8.0...v5.0.0

I do not know what tokens the `alaska-` elements are looking for. Regardless, you CAN load both `--auro-` and `--ds-` tokens in the same app, and there are no side effects that I am aware of. 

All of the latest versions of Auro components will be looking for the new `--ds-` tokens and WCSS 5.x.

### Steps to resolve would be; 

Update to the latest version of `@aurodesignsystem/design-tokens` and `@alaskaairux/design-tokens`, see the following code example

```js
// -=-=-=-=-=-=-=-=-=-=- LEGACY =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
// Required to support components not yet updated to new lib
// -=-=-=-=-=-=-=-=-=-=- LEGACY =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
import '@alaskaairux/design-tokens/dist/tokens/CSSCustomProperties.css';


// Current auro tokens
import '@aurodesignsystem/design-tokens/dist/tokens/CSSCustomProperties.css';
```

Update to the latest version of WCSS. The issue you see in the console is a conflict between versions of peerDepednencies. Use the `--legacy-peer-deps` flag and that will temp resolve the situation. 

As you update more and more element dependencies you will see that go away.

**Reply by blackfalcon** [reply]
All recently published components should have updated dependencies. Including auro-badge mentioned in this question https://github.com/AlaskaAirlines/auro-badge/blob/main/package.json#L28-L31

All dependencies are listed with a `^` operator. Theoretically, all users should be able to perform a `npm ci` command that will complete a clean install of all dependencies at their latest versions and dedupe as needed.
